src/main/java/com/vg/hm/samples/usermanagementsvc/util/VehicleAPIRetrofitClient.java

```diff
@@ -3,30 +3,65 @@
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import com.vg.hm.samples.usermanagementsvc.service.VehicleAPIService;
+import lombok.SneakyThrows;
+import lombok.extern.log4j.Log4j2;
+import okhttp3.*;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import retrofit2.Retrofit;
 import retrofit2.converter.gson.GsonConverterFactory;
 
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+@Log4j2
 @Configuration
 public class VehicleAPIRetrofitClient {
     @Value("${vehicleAPI.baseUrl}")
     private String BASE_URL;
 
     @Bean(name = "vehicleServiceRetrofitClient")
     public VehicleAPIService getRetrofit() {
-
         Gson gson = new GsonBuilder()
                 .setDateFormat("yyy-MM-dd'T'HH:mm:ssz")
+                .setLenient()
                 .create();
 
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        builder.addInterceptor(responseInterceptor);
+        OkHttpClient client = builder.build();
+
         Retrofit retrofit = new Retrofit.Builder()
                 .baseUrl(BASE_URL)
                 .addConverterFactory(GsonConverterFactory.create(gson))
+                .client(client)
                 .build();
-        VehicleAPIService client = retrofit.create(VehicleAPIService.class);
-        return client;
+        return retrofit.create(VehicleAPIService.class);
+    }
+
+    private Interceptor responseInterceptor = new Interceptor() {
+    @SneakyThrows
+    @Override
+    public Response intercept(Chain chain) throws IOException {
+        log.info("Intercepting http call");
+        Request request = chain.request();
+        Response response = chain.proceed(request);
```

Comment:
This is wrong. You are intercepting the API call, but you are also making the backend call and then manipulating the response to return a different/mocked response. Ideally, your interceptor should prevent making a call to your backend, and instead returning a mocked json. You should only do "chain.proceed" if we don't want to mock a response.  

To make things easier, create a class called "MockRetrofitInterceptor" and extends from Interceptor class. Basically move your code from the line 45-60 to your new class.

Then, on your intercept(Chain chain) method, you can read headers from the request, like this:
`Request request = chain.request();
Headers headers = request.headers();`

Then if a special header like "enable-mocks" is present on the list of headers, then you can execute your logic to return a mocked json. If the header is not present, then you should just do:
`return chain.proceed(request)`
This is to exit from the method.